substrate/client/network/sync/src/strategy/chain_sync.rs

```diff
@@ -337,6 +337,9 @@ pub struct ChainSync<B: BlockT, Client> {
 	import_existing: bool,
 	/// Block downloader
 	block_downloader: Arc<dyn BlockDownloader<B>>,
+	/// Is block pruning enabled? This indicates that the user is not
+	/// interested in historical blocks, so gap sync will be skipped.
+	block_pruning_enabled: bool,
```

Comment:
This component should now know about blocks pruning IMO. The naming should reflect whether gap sync fetches bodies and justifications or not.